### DISCUSSION TITLE
Refinement Module in E2E-VarNet Implementation

### DISCUSSION THREAD

**Discussion by michael-s-yao** [facebookresearch/fastMRI#229] [Q&A] [answered]
Hello, I am reading through the [2020 paper](https://arxiv.org/abs/2004.06688) from Sriram A et al on their work with E2E-VarNet and have a question on [this line](https://github.com/facebookresearch/fastMRI/blob/6683d2c046d4e147f9fe39afc10416c9f9332430/fastmri/models/varnet.py#L322) in the fastMRI repository. Specifically, line 10 of their paper suggests that the refinement module output should be added to the previous predicted kspace, whereas the implementation in the repository seems like the refinement module output is being subtracted. Just want to check to see which approach is intended and if I'm thinking about their implementation correctly. Thank you in advance!

**Comment by mmuckley** [comment, marked as answer]
Hello @michael-s-yao, it looks like this formulation is in the original open-sourcing by the authors [here](https://github.com/facebookresearch/fastMRI/blob/8abf45009f59f481db35eb2bd53d627f2c4ad6f7/models/varnet/varnet.py). The private internal code used for this paper also follows this convention, so I think equation (10) in the paper has a typo.

CC @anuroopsriram

**Reply by michael-s-yao** [reply]
Got it, thanks @mmuckley for the explanation!
